### DISCUSSION TITLE
ui5-webcomponents-sample-lit

### DISCUSSION THREAD

**Discussion by nemethmik** [UI5/webcomponents#4057] [Show and tell]
I have completed a three part series on re-making a [ui5-webcomponents-sample-lit](https://github.com/nemethmik/ui5-webcomponents-sample-lit) using ViteJS, Lit, and TypeScript. I started from the React sample; in the first iteration I simply converted it keeping the "React mindset". In the next iteration I dropped the myriads of callbacks, and replaced them with the clean browser-standard Custom Event notification machinery applying a **reducer-like dispatching and state management solution**. In the third iteration I introduced MobX app state management to separate the UI app component from the app state. 
I am still pretty confident that UI5 web components are (almost) ready for heavy-demand production. I highly recommend to make typings part of the release. I can live without typings, but other TypeScript-fanatic workshops would disqualify. The other web component libraries I have experimenting with all have decent typings. 
What is especially fascinating with this toolset is that all components are web-standard: template literals, classes and object, custom elements, slots, shadow DOM for encapsulation. No need for proprietary templating, component machinery, callback wizardry, render-props and arcane hooks. TypeScript is brutally important for code maintainability. ViteJs is fantastic, blazing fast. Lit/Lit-HTML is a joy to use, MobX is dead easy.  I am blown with this toolset. 
So, congratulations to the UI5 Web Component authors, great product, and thank you very much for the help. 
PS: You can check out the [online app on Azure](https://orange-field-01bd0ee03.azurestaticapps.net/).

**Comment by fifoosid** [comment]
Hi @nemethmik 

We are glad that you are happy with the stack that includes UI5 Web Components, The application that you have built looks & performs great! I will put a link to the [source code](https://github.com/nemethmik/ui5-webcomponents-sample-lit) in case someone needs it in the future.

When it comes to the TypeScript typings, It might be a good idea to fire a [feature request](https://github.com/SAP/ui5-webcomponents/issues/new?assignees=&labels=&template=feature_request.md), where you will be able to keep track of the progress.

Besides, as you might already know, we are planing on releasing our first stable version(1.0.0) soon. Stay tuned!
